pkgs/build-support/coq/meta-fetch/default.nix

```diff
@@ -1,7 +1,6 @@
 { lib, stdenv, fetchzip }@args:
-let lib' = lib; in
-let lib = import ../extra-lib.nix {lib = lib';}; in
-with builtins; with lib;
+let lib' = import ../extra-lib.nix {inherit lib;}; in
+with builtins; with lib';
```

Comment:
How about not using `with` for such a large scope at all? See:
https://nix.dev/anti-patterns/language.html#with-attrset-expression